src/WebJobs.Script.Scaling/IWorkerTable.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Script.Scaling
+{
+    /// <summary>
+    /// worker table implementation providing CRUD over the table
+    /// </summary>
+    public interface IWorkerTable
+    {
+        /// <summary>
+        /// enumerate all workers in the table
+        /// </summary>
+        Task<IEnumerable<IWorkerInfo>> List();
+
+        /// <summary>
+        /// add new worker to the table
+        /// </summary>
+        Task AddOrUpdate(IWorkerInfo worker);
+
+        /// <summary>
+        /// delete worker from the table
+        /// </summary>
+        Task Delete(IWorkerInfo worker);
+
+        /// <summary>
+        /// acquire lock on the table
+        /// </summary>
+        Task<IWorkerTableLock> AcquireLock();
+
+        /// <summary>
+        /// get the current manager
+        /// </summary>
+        Task<IWorkerInfo> GetManager();
```

Comment:
This API and the SetManager API below are strange. The don't return/set a manager, they operate on an IWorkerInfo